src/crop.sol

```diff
@@ -102,41 +102,41 @@ contract CropJoin {
         return sub(bonus.balanceOf(address(this)), stock);
     }
 
-    function harvest() internal {
+    function harvest(address from, address to) internal {
         if (total > 0) share = add(share, rdiv(crop(), total));
 
-        uint256 last = crops[msg.sender];
-        uint256 curr = rmul(stake[msg.sender], share);
-        if (curr > last) require(bonus.transfer(msg.sender, sub(curr, last)));
+        uint256 last = crops[from];
+        uint256 curr = rmul(stake[from], share);
+        if (curr > last) require(bonus.transfer(to, sub(curr, last)));
         stock = bonus.balanceOf(address(this));
     }
 
-    function join(uint256 val) public virtual {
+    function join(address usr, uint256 val) public virtual {
```

Comment:
This is no fully-satisfactory solution for naming here as far as I can see, but I would suggest calling the address here `urn` instead of `usr`; it's more consistent with what we tend to do in the [other gem joins](https://github.com/makerdao/dss-gem-joins/tree/master/src), and more importantly having a different name that what's used in `exit` helps a reader of the code realize that the address parameters in `join` and `exit` are not equivalent in important ways.

For the parameter in `exit`, you might consider `guy` instead of `usr` (`urn` and `usr` are pretty similar, and in `dss-gem-joins` it seems `guy` is the more commonly-used name, although it's not uniform).